gdal_subsetter/transform.py

```diff
@@ -251,32 +279,39 @@ def process_netcdf(self, source, basename, input_filename, output_dir, layername
 
         return layernames, result, process_msg
 
-    def process_zip(self, source, basename, input_filename, output_dir, layernames):
-
-        result = None
-
-        variables = source.process('variables')
-
-        variableslist = [item.name for item in variables]
-
-        [is_tif, tiffile, msg_tif, is_nc, ncfile, msg_nc] = self.unpack_zipfile(input_filename, output_dir, variableslist)
+    def process_zip(self, source: HarmonySource, basename: str,
+                    zipfile_name: str, output_dir: str, layernames: List[str]):
+        """ Unpack the input zip file and process the individual files it
+            contains. If there are GeoTIFF files, only these will be
+            processed.
 
-        if is_tif:
-            if tiffile:
-                layernames, result = self.process_geotiff(
-                    source, basename, tiffile, output_dir, layernames)
+            `HarmonyAdapter.unpack_zipfile` will attempt to stack multiple
+            GeoTIFFs contained within the zip file, and process those as a
+            single file. The possibility of multiple NetCDF-4 files does not
+            seem to be accounted for here.
 
-            return layernames, result, msg_tif
-
-        elif is_nc:
-            if ncfile:
-                layernames, result = self.process_netcdf(source, basename,
-                                                         ncfile, output_dir,
-                                                         layernames)
-
-            return layernames, result, msg_nc
+        """
+        variable_names = [item.name for item in source.process('variables')]
+        tiffile, msg_tif, ncfile, msg_nc = self.unpack_zipfile(zipfile_name,
+                                                               output_dir,
+                                                               variable_names)
```

Comment:
I've spent a little while puzzling over `ncfile`. I am reasonably sure that will always be a list (because `HarmonyAdapter.unpack_zip` is calling `get_files_from_unpackzipfiles`, which returns a list). So it seems to me that:

a) This code has never been run on a zipfile containing NetCDF-4 file(s).
b) This code should probably break in such an instance - because `HarmonyAdapter.process_netcdf` is expecting a file name, not a list of them (even a single element list).

Part of me is still holding out hope that we just drop NetCDF-4 support in this service. but otherwise, I think we need to:

a) either loop through `ncfile` (along with giving it a better variable name).
b) or assume zip files only contain GeoTIFFs, and then not worry about zipped NetCDF-4 files.

I'd love someone to sanity check this understanding, though.

Reply:
So I agree this has never been run with a list of netcdf files, at least not successfully. 

I think I agree that we are trying to not add aggregation of netcdf file outputs to this service. If so, let's just error if a zip contains multiple netcdf files.

Reply:
I've updated this function to do the following:

1) If there are multiple NetCDF-4 files in the zip file, an exception is raised.
2) If there is a single NetCDF-4 file in the zip file, the first element of the `ncfile` list is sent along to `HarmonyAdapter.process_netcdf`.